src/test/java/com/apextrio/peak/PeakApplicationTests.java

```diff
@@ -1,17 +1,319 @@
+			this.mvc.perform(post("/teams/3")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/teams/3"));
+		} catch (java.lang.Exception e) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	public void testPostTeamsIdNoAuth() {
+		try {
+			this.mvc.perform(post("/teams/3")).andExpect(status().isForbidden()).andExpect(redirectedUrl("/error"));
+		} catch (java.lang.Exception e) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	@WithUserDetails("NJCrain")
+	public void	testDeleteTeamId() {
+		try {
+			this.mvc.perform(delete("/teams/2")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/teams/2"));
+		} catch (java.lang.Exception e) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	@WithUserDetails("jasonb315")
+	public void	testDeleteTeamIdLastUser() {
+		try {
+			this.mvc.perform(delete("/teams/2")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"));
+		} catch (java.lang.Exception e) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	public void testPostSignup() {
+		try {
+			this.mvc.perform(post("/signup?username=test&password=123456&firstName=test&lastName=test&bio=test")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/"));
+		} catch (java.lang.Exception e ) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	public void testPostSignupWithTeamId() {
+		try {
+			this.mvc.perform(post("/signup?teamId=1&username=test&password=123456&firstName=test&lastName=test&bio=test")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/teams/1"));
+		} catch (java.lang.Exception e ) {
+			System.err.println(e);
+		}
+	}
+
+	@Test
+	@WithUserDetails("NJCrain")
+	public void testPostUpdateProfile() {
+		try {
+			this.mvc.perform(post("/updateProfile?bioUpdate=test")).andExpect(status().is3xxRedirection()).andExpect(redirectedUrl("/myProfile"));
+		} catch (java.lang.Exception e) {
+			System.err.println(e);
+		}
+	}
+
+	//------------
+	//CRUD TESTING
+	//------------
+
+	@Test
+	public void testCreateUser() {
+		AppUser testUser = new AppUser();
+		testUser = userRepo.save(testUser);
+		assertNotNull(testUser);
+		assertTrue(testUser.getId() != 0);
+	}
+
+	@Test
+	public void testCreateTeam() {
+		Team testTeam = new Team();
+		testTeam = teamRepo.save(testTeam);
+		assertNotNull(testTeam);
+		assertTrue(testTeam.getId() != 0);
+	}
+
+	@Test
+	public void testReadUser() {
+		AppUser user = userRepo.findByUsername("NJCrain");
+		assertNotNull(user);
+		assertEquals("test", user.getBio());
+		assertEquals("Nick", user.getFirstName());
+		assertEquals("Crain", user.getLastName());
+	}
+
+	@Test
+	public void testReadTeam() {
+		Team testTeam = teamRepo.findById( (long) 3).get();
+		assertEquals("Team 2-2", testTeam.getName());
+		assertEquals(3, testTeam.getDifficulty());
+		assertEquals("Descriptions are lame", testTeam.getDescription());
+		assertEquals(3, testTeam.getCapacity());
+	}
+
+	@Test
+	public void testReadResort() {
+		Resort r = resortRepo.findById((long) 1).get();
```

Comment:
Assuming the DB is already filled in with exactly the right data, at exactly the right ID, makes me nervous. I'd prefer to test this by adding something and then reading it.